doc/en/user/source/rest/api/stylegenerator.rst

```diff
@@ -0,0 +1,125 @@
+.. _rest_api_stylegenerator:
+
+Style Generator
+===============
+
+The Style Generator endpoint is used for generating default styles. Styles generated using this endpoint may be added to geoserver using the :ref:`styles <rest_api_styles>` endpoint.
+
+``/stylegenerator[.<format>]``
+------------------------------
```

Comment:
For integrating with the styles endpoint, I was mostly concerned about
colliding with the existing endpoints.

I do like your PUT idea however Jody, it integrates much better with the
existing API - in this case I guess the template to be used would be passed
as a query paramete (defaulting to generic)?

Torben

On Thu, Aug 13, 2015 at 1:14 PM, Jody Garnett <notifications@github.com>
wrote:

> In doc/en/user/source/rest/api/stylegenerator.rst
> <https://github.com/geoserver/geoserver/pull/1182#discussion_r37019825>:
>
> > @@ -0,0 +1,125 @@
> > +.. _rest_api_stylegenerator:
> > +
> > +Style Generator
> > +===============
> > +
> > +The Style Generator endpoint is used for generating default styles. Styles generated using this endpoint may be added to geoserver using the :ref:`styles <rest_api_styles>` endpoint.
> > +
> > +``/stylegenerator[.<format>]``
> > +------------------------------
>
> Interesting, I may of done this as an endpoint on the styles REST API
> endpoint.
>
> Something like PUT /workspaces//styles/[.] would complete the table?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoserver/geoserver/pull/1182/files#r37019825>.
>